### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by TacoTheDank**
## Dependency changes
- gradle 4.6 -> 4.10.3
- realm-gradle-plugin 5.1.0 -> 5.8.0
- coveralls-gradle-plugin 2.4.0 -> 2.8.2
- fastscroll 0.2.4 -> 0.2.5
- okhttp 3.10.0 -> 3.12.1
- retrofit 2.4.0 (and 2.2.0) -> 2.5.0
- Note that jakewharton's retrofit2-rxjava2-adapter is deprecated, use retrofit's adapter-rxjava2 instead
- flowlayout 0.4.0 -> 0.4.1
- rxandroid 2.0.2 -> 2.1.0
- rxjava 2.1.17 -> 2.2.4
- gson 2.8.2 -> 2.8.5
- licensesdialog 1.8.1 -> 1.9.0
- junit 4.12 -> 4.13-beta-1
- mockito-core 2.20.0 -> 2.23.4

## Code Changes
- Optimize some code
- Add badges to README.md
- Reformat code
- Change app name from /r/Android App store to /r/Android App Store
- Migrate app to appCompat

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,15 @@
 
 The official app for displaying the curated app list from the [/r/Android wiki](https://www.reddit.com/r/android/wiki/apps), inspired by [this reddit post](https://redd.it/50rafp).
 
+<a href="https://f-droid.org/packages/subreddit.android.appstore">
+    <img src="https://fdroid.gitlab.io/artwork/badge/get-it-on.png"
+         alt="Get it on F-Droid" height="80">
```
Comment (bobheadxi) [inline README.md:12]: 😍

Comment (TacoTheDank) [reply README.md:12]: xd what?

**Comment by d4rken** [comment]
The PR is too huge, too many files touched, difficult to review. :frowning: 

I did a quick glance and most looks okay, except for the autoformat. Don't just run autoformat for a PR on a whole project. :raised_hand_with_fingers_splayed: . Can you undo it?

File: app/src/main/java/subreddit/android/appstore/backend/github/GithubApi.java [hunk 2]
```diff
@@ -18,21 +18,28 @@
     Observable<List<Contributor>> getContributors();
 
     class Release {
-        @SerializedName("url") public String releaseUrl;
-        @SerializedName("tag_name") public String tagName;
-        @SerializedName("name") public String releaseName;
-        @SerializedName("body") public String releaseDescription;
+        @SerializedName("url")
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/github/GithubApi.java:21]: e.g. bad autoformat, previously was more readable.

File: app/src/main/java/subreddit/android/appstore/backend/reddit/TokenRepository.java [hunk 3]
```diff
@@ -63,7 +63,8 @@ public TokenRepository(Context context,
         TokenApi.Token token = null;
         try {
             token = gson.fromJson(preferences.getString(PREF_KEY, null), TokenApi.Token.class);
-        } catch (JsonSyntaxException ignore) {}
+        } catch (JsonSyntaxException ignore) {
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/reddit/TokenRepository.java:66]: Bad autoformat

File: app/src/main/java/subreddit/android/appstore/backend/reddit/wiki/LiveWikiRepository.java [hunk 4]
```diff
@@ -75,18 +74,21 @@ public void refresh() {
                                                 String name = r.getAppName();
                                                 if (name.contains("&")) {
                                                     int pos = name.indexOf("&") + 1;
-                                                    name = name.substring(0, pos) + "amp;" + name.substring(pos, name.length());
+                                                    name = name.substring(0, pos) + "amp;"
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/reddit/wiki/LiveWikiRepository.java:77]: Bad autoformat

File: app/src/main/java/subreddit/android/appstore/backend/reddit/wiki/caching/WikiDiskCache.java [hunk 5]
```diff
@@ -55,7 +52,8 @@ public void putAll(Collection<AppInfo> appInfos) {
                 .map(cachedAppInfos -> {
                     Collection<AppInfo> appInfos = new ArrayList<>();
                     Timber.d("Returned %d AppInfos from cache", cachedAppInfos.size());
-                    for (CachedAppInfo cachedAppInfo : cachedAppInfos) appInfos.add(cachedAppInfo.toAppInfo(gson));
+                    for (CachedAppInfo cachedAppInfo : cachedAppInfos)
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/backend/reddit/wiki/caching/WikiDiskCache.java:55]: Bad autoformat.

Either single line loops, or loop with braces.

File: app/src/main/java/subreddit/android/appstore/screens/navigation/CategoryFilter.java [hunk 6]
```diff
@@ -96,13 +97,13 @@ public boolean equals(Object o) {
 
         CategoryFilter that = (CategoryFilter) o;
 
-        if (primaryCategory==null ? that.primaryCategory!=null : !primaryCategory.equals(that.primaryCategory))
+        if (primaryCategory == null ? that.primaryCategory != null : !primaryCategory.equals(that.primaryCategory))
```
Comment (d4rken) [inline app/src/main/java/subreddit/android/appstore/screens/navigation/CategoryFilter.java:100]: Good autoformat

File: app/src/main/res/drawable/ic_close_white_24dp.xml [hunk 7]
```diff
@@ -1,9 +1,9 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-        android:width="24dp"
-        android:height="24dp"
-        android:viewportWidth="24.0"
-        android:viewportHeight="24.0">
+    android:width="24dp"
```
Comment (d4rken) [inline app/src/main/res/drawable/ic_close_white_24dp.xml:2]: Questionable autoformat. Not bad, but theoretically just noise and not worth a commit.

**Comment by d4rken** [comment]
Added some examples on autoformat issues.

Does anyone know why Travis isn't passing?

Happy holidays btw :tada:

**Comment by TacoTheDank** [comment]
Will make changes, merry holichristmas happy days.

I'll just undo all the autoformats.

**Pull request closed without merging**